substrate/frame/revive/src/exec.rs

```diff
@@ -825,9 +826,15 @@ where
 		)?
 		.expect(FRAME_ALWAYS_EXISTS_ON_INSTANTIATE);
 		let address = T::AddressMapper::to_address(&stack.top_frame().account_id);
-		stack
+		let result = stack
 			.run(executable, input_data)
-			.map(|_| (address, stack.first_frame.last_frame_output))
+			.map(|_| (address, stack.first_frame.last_frame_output));
+		if let Ok((contract, ref output)) = result {
+			if !output.did_revert() {
+				Contracts::<T>::deposit_event(Event::Instantiated { deployer, contract });
+			}
+		}
+		result
```

Comment:
I added this within `exec`, in `run_instantiate` which is called only from the top frame. The other possibility was to have this in `bare_instantiate` of the pallet's lib.rs, but I thought it would be better to have it in here since this is the function performing the instantiation (and some tests are actually calling this function directly).